db/migrate/20160513163056_deduplicate_links.rb

```diff
@@ -0,0 +1,30 @@
+class DeduplicateLinks < ActiveRecord::Migration
+  def up
+    res = Link.connection.execute(<<-SQL
+      SELECT SUM(cnt), COUNT(cnt)
+      FROM (
+        SELECT
+          link_set_id,
+          target_content_id,
+          link_type,
+          COUNT(id) cnt
+        FROM links
+        WHERE target_content_id IS NOT NULL
+        GROUP BY 1,2,3
+        HAVING COUNT(id) > 1
+      ) sub
+      SQL
+    )
+    puts "#{res[0]["count"]} unique records of #{res[0]["sum"]} total duplicated"
+
+    res = Link.connection.execute(<<-SQL
+      DELETE FROM links USING links l2
+        WHERE links.link_set_id = l2.link_set_id
+        AND links.target_content_id = l2.target_content_id
+        AND links.link_type = l2.link_type
+        AND links.id < l2.id
```

Comment:
Interpreting this out loud using my rusty SQL comprehension:

Deletes all the items from the `links` table that have those 3 columns duplicated in another item with a higher `id`. Because the primary key is sequential and auto-incremented (as is convention), this works out to leave at least one of each, the "newer" (higher `id`) one.

Seems sane to me. 👌